src/Illuminate/Queue/QueueManager.php

```diff
@@ -33,15 +34,24 @@ class QueueManager implements FactoryContract, MonitorContract
      */
     protected $connectors = [];
 
+    /**
+     * The shared data that should be added to the queue payload.
+     *
+     * @var \Illuminate\Support\Collection
+     */
+    protected $shared;
+
     /**
      * Create a new queue manager instance.
      *
      * @param  \Illuminate\Foundation\Application  $app
+     * @param  \Illuminate\Support\Collection  $shared
      * @return void
      */
-    public function __construct($app)
+    public function __construct($app, Collection $shared = null)
```

Comment:
Why forcing $shared to be a Collection?

Reply:
As shared needs to be a collection so I can easily add things to it when they use Queue::share. It only really exists in a construction argument for easier testing and if people use queues outside of laravel

Reply:
Other problem is that this line does not make any sense, Collection $shared = null will fail if you do not pass the second argument.
I would better remove type hinting here and change it to $shared=[]

Reply:
You don't need to pass the second argument over. As below if it is null which is the default  it will create a new collection. I don't want it to be a straight array as collections are easier to deal with and have nice helper methods on them.

Reply:
But that is my point, it will not, if i do something like
```
$manager = new QueueManager($app); // Assuming I do not need $shared as it has a default value
```
This will fail and tell me 'Expected instance of Collection null given" (or something like that)

Reply:
No it wouldn't - http://sandbox.onlinephpfunctions.com/code/278cc7abf9638c7787b6d9091e71e4646a9b2574 execute that code.

Also Example 9 on here - http://php.net/manual/en/functions.arguments.php